docs/data/date-pickers/custom-field/custom-behavior/MaskedMaterialTextField.tsx

```diff
@@ -0,0 +1,168 @@
+
+  const { forwardedProps, internalProps } = useSplitFieldProps(other, 'date');
+
+  const {
+    format,
+    value: valueProp,
+    defaultValue,
+    onChange,
+    timezone,
+    onError,
+  } = internalProps;
+
+  const [value, setValue] = useControlled({
+    controlled: valueProp,
+    default: defaultValue ?? null,
+    name: 'MaskedField',
+    state: 'value',
+  });
+
+  // Control the input text
+  const [valueStr, setValueStr] = React.useState<string>(() =>
+    getValueStrFromValue(value, format),
+  );
+
+  React.useEffect(() => {
+    if (value && value.isValid()) {
+      const newDisplayDate = getValueStrFromValue(value, format);
+      setValueStr(newDisplayDate);
+    }
+  }, [format, value]);
+
+  const parsedFormat = useParsedFormat(internalProps);
+
+  const { hasValidationError, getValidationErrorForNewValue } = useValidation({
+    value,
+    timezone,
+    onError,
+    props: internalProps,
+    validator: validateDate,
+  });
+
+  const handleValueStrChange = (newValueStr: string) => {
+    setValueStr(newValueStr);
+
+    const newValue = dayjs(newValueStr, format);
+    setValue(newValue);
+
+    if (onChange) {
+      onChange(newValue, {
+        validationError: getValidationErrorForNewValue(newValue),
+      });
+    }
+  };
+
+  const rifmFormat = React.useMemo(() => {
+    const formattedDateWith1Digit = staticDateWith1DigitTokens.format(format);
+    const inferredFormatPatternWith1Digits = formattedDateWith1Digit.replace(
+      ACCEPT_REGEX,
+      MASK_USER_INPUT_SYMBOL,
+    );
+    const inferredFormatPatternWith2Digits = staticDateWith2DigitTokens
+      .format(format)
+      .replace(ACCEPT_REGEX, '_');
+
+    if (inferredFormatPatternWith1Digits !== inferredFormatPatternWith2Digits) {
+      throw new Error(
+        `Mask does not support numbers with variable length such as 'M'.`,
+      );
+    }
+
+    const maskToUse = inferredFormatPatternWith1Digits;
+
+    return function formatMaskedDate(valueToFormat: string) {
+      let outputCharIndex = 0;
+      return valueToFormat
+        .split('')
+        .map((char, inputCharIndex) => {
+          ACCEPT_REGEX.lastIndex = 0;
+
+          if (outputCharIndex > maskToUse.length - 1) {
+            return '';
+          }
+
+          const maskChar = maskToUse[outputCharIndex];
+          const nextMaskChar = maskToUse[outputCharIndex + 1];
+
+          const acceptedChar = ACCEPT_REGEX.test(char) ? char : '';
+          const formattedChar =
+            maskChar === MASK_USER_INPUT_SYMBOL
+              ? acceptedChar
+              : maskChar + acceptedChar;
+
+          outputCharIndex += formattedChar.length;
+
+          const isLastCharacter = inputCharIndex === valueToFormat.length - 1;
+          if (
+            isLastCharacter &&
+            nextMaskChar &&
+            nextMaskChar !== MASK_USER_INPUT_SYMBOL
+          ) {
+            // when cursor at the end of mask part (e.g. month) prerender next symbol "21" -> "21/"
```

Comment:
could we also move the cursor to behind the '/'?
Right now it looks like this: `21|/`, but preferably it would do this instead: `21/|`

Reply:
In v5 the cursor was before the `/`
I do agree that having it after would be better, but I did not find a solution to do it :grima

Reply:
Yeah, i thought so ... its not blocking IMO